src/main/java/lotto/domain/winning/WinningChecker.java

```diff
@@ -11,33 +11,26 @@
  */
 public class WinningChecker {
 
-    private final List<Integer> winningNumbers;
-    private static final WinningStatistics winningStatistics = new WinningStatistics();
+    private final WinningNumber winningNumber;
 
     private WinningChecker(WinningNumber winningNumber) {
-        this.winningNumbers = winningNumber.value();
+        this.winningNumber = winningNumber;
     }
 
     public static WinningChecker of(WinningNumber winningNumber) {
         return new WinningChecker(winningNumber);
     }
 
     public WinningStatistics winningStatistics(Lottos lottos) {
+        final WinningStatistics winningStatistics = new WinningStatistics();
         for (Lotto lotto : lottos.list()) {
             winningStatistics.add(getWinningResult(lotto));
         }
```

Comment:
```suggestion
        //아래 로직들이 있으면 적절한 클래스와 메서드를 고민해 보시죠!
        Map<WinningRank, Long> countPerRank = lottos.list()
            .stream()
            .map(this::getWinningResult)
            .collect(Collectors.groupingBy(Function.identity(),
                                            Collectors.counting()));

        Map<WinningRank, Integer> winningResult = Arrays.stream(WinningRank.values())
            .filter(rank -> rank != WinningRank.NONE)
            .collect(Collectors.toMap(Function.identity(),
                rank -> Math.toIntExact(countPerRank.get(rank))));
```
WinningStatistics를 View에서 꺼내 쓸 때 unmodifiableMap으로 변경 해주신 걸 보니 불변성에 대해 고민해주셨네요 👍 👍 
한가지 아쉬운점은 현재 구조에서는 winningStatistics 메서드 밖에서도 얼마든지 당첨 결과에대한 조작이 가능한 구조인데요 😭
불변성에 대해 고민해주셨으니 한 번 객체를 생성하면 추가로 변경할 수 없는 방향으로 바꿔보시는건 어떨까요? 😊

코멘트 드린 로직들을 적절한 클래스의 메서드로 옮겨 주시면 객체를 생성하면서 바로 unmodifiableMap을 멤버변수에 할당 할 수 있을 것 같습니다 :)

Reply:
https://github.com/next-step/java-lotto/pull/987/commits/962bffdb88653b3a6a0ba6d5b5f21705f95bd492 에 반영완료